Make block import a shared component and use it in domain block processor.

vedhavyas opened the pull request:
Turns out domain block processor is using client's BlockImport instead of the provided block import. I have made DomainBlockImport as shared between all the required tasks. If you think this could be simplified, open to suggestions.

### Code contributor checklist:
* [x] I have read, understood and followed [contributing guide](https://github.com/subspace/subspace/blob/main/CONTRIBUTING.md)

liuchengxu reviewed (commented):
Why this PR is necessary? We had a TODO (at line 31 in import_queue.rs) to remove the `DomainBlockImport` and probably even the entire `consensus-relay-chain` actually as it's currently unneeded and technically a leftover from the original cumulus.

liuchengxu commented on domains/client/domain-executor/src/system_bundle_processor.rs:
```diff
@@ -28,13 +28,14 @@ where
+    for SystemBundleProcessor<Block, PBlock, Client, PClient, Backend, E, BI>
 where
     Block: BlockT,
+    Backend: sc_client_api::Backend<Block>,
```
```suggestion
```

liuchengxu commented on domains/client/domain-executor/src/system_executor.rs:
```diff
@@ -35,13 +35,14 @@ where
+    for Executor<Block, PBlock, Client, PClient, TransactionPool, Backend, E, BI>
 where
     Block: BlockT,
+    Backend: sc_client_api::Backend<Block>,
```
```suggestion
```

liuchengxu commented on domains/client/domain-executor/src/domain_block_processor.rs:
```diff
@@ -30,21 +30,23 @@ where
-    backend: Arc<Backend>,
-    fraud_proof_generator: FraudProofGenerator<Block, PBlock, Client, PClient, Backend, E>,
-    domain_confirmation_depth: NumberFor<Block>,
+    pub(crate) domain_id: DomainId,
+    pub(crate) client: Arc<Client>,
+    pub(crate) primary_chain_client: Arc<PClient>,
+    pub(crate) primary_network_sync_oracle: Arc<dyn SyncOracle + Send + Sync>,
+    pub(crate) backend: Arc<Backend>,
+    pub(crate) fraud_proof_generator:
+        FraudProofGenerator<Block, PBlock, Client, PClient, Backend, E>,
+    pub(crate) domain_confirmation_depth: NumberFor<Block>,
+    pub(crate) block_import: Arc<BI>,
```
Are you trying to avoid the complaint from Clippy that there will be too many arguments?

vedhavyas replied:
yeap. Did not want to add an expception

liuchengxu commented on domains/client/domain-executor/src/core_bundle_processor.rs:
```diff
@@ -36,13 +45,14 @@ where
+    for CoreBundleProcessor<Block, SBlock, PBlock, SClient, Client, PClient, Backend, E, BI>
 where
     Block: BlockT,
+    Backend: sc_client_api::Backend<Block>,
```
```suggestion
```

liuchengxu reviewed (commented):
This PR makes sense now. Aside from some nits, it looks good to me once tests pass. In addition, I believe a bunch of docs around `DomainBlockImport` need to be updated since it's activated now (not necessarily in this PR).

vedhavyas committed:
Make block import a shared component and use it in domain block processor.

Turns out domain block processor is using client's BlockImport instead of the provided block import. I have made DomainBlockImport as shared between all the required tasks. If you think this could be simplified, open to suggestions.

vedhavyas committed:
remove Custom fork choice as we need to import as is instead of overriding it

NingLin-P reviewed (approved):
Make sense overall!

vedhavyas committed:
Merge branch 'main' into domain/client_bi
